data_management/models.py

```diff
@@ -645,26 +648,44 @@ class ExternalObject(BaseModel):
     `last_updated`: Datetime that this record was last updated
 
     `updated_by`: Reference to the user that updated this record
+
+    `version`: Version identifier of the `DataProduct` associated with this `ExternalObject`
     """
-    ADMIN_LIST_FIELDS = ('doi_or_unique_name', 'title')
+    ADMIN_LIST_FIELDS = ('other_unique_name', 'title', 'version')
 
     data_product = models.ForeignKey(DataProduct, on_delete=models.PROTECT, related_name='external_objects')
-    doi_or_unique_name = models.CharField(max_length=CHAR_FIELD_LENGTH, null=False, blank=False)
+    other_unique_name = models.CharField(max_length=CHAR_FIELD_LENGTH, null=True, blank=True)
+    identifier = models.URLField(max_length=TEXT_FIELD_LENGTH, null=True, unique=True, blank=True)
     primary_not_supplement = models.BooleanField(default=True)
     release_date = models.DateTimeField()
     title = models.CharField(max_length=CHAR_FIELD_LENGTH)
     description = models.TextField(max_length=TEXT_FIELD_LENGTH, null=True, blank=True)
+    version = VersionField(null=True, blank=True, editable=False)
     original_store = models.ForeignKey(StorageLocation, on_delete=models.PROTECT, related_name='original_store_of', null=True, blank=True)
 
     class Meta:
         constraints = [
             models.UniqueConstraint(
-                fields=('doi_or_unique_name', 'title'),
+                fields=('other_unique_name', 'identifier', 'title', 'version'),
                 name='unique_external_object'),
+            models.CheckConstraint(
+                name="%(app_label)s_%(class)s_identifier_or_other_unique_name",
+                check=(
+                    models.Q(identifier__isnull=True, other_unique_name__isnull=False)
+                    | models.Q(identifier__isnull=False, other_unique_name__isnull=True)
+                    | models.Q(identifier__isnull=False, other_unique_name__isnull=False)
```

Comment:
I think @alahiff is probably right that we should only allow `identifier` or `other_unique_name`, not both.

Reply:
It's actually my fault that Nathan added that :-)

Reply:
removing line 676 will go back to only allowing one or the other, but not both. I'll do that now 👍